### PR TITLE
chore: Transaction refactoring

### LINKED ISSUE #14004: Refactor transactions
`StateSignatureTransaction` & `SwirldTransaction` classes should be removed since they are not needed anymore. `ConsensusTransactionImpl` should become `PayloadWrapper`. Apart from the 2 mutable fields it already has, it should also hold the payload.

### PR DESCRIPTION
**Pull request by timo0**
Fix #14004

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/transaction/PayloadWrapper.java [hunk 1]
```diff
@@ -50,6 +61,29 @@ public void setConsensusTimestamp(@NonNull final Instant consensusTimestamp) {
...
+        return payload;
+    }
+
+    @Override
+    public int getSize() {
+        if (PayloadOneOfType.STATE_SIGNATURE_PAYLOAD.equals(payload.kind())) {
+            final StateSignaturePayload stateSignaturePayload = payload.as();
+            return Long.BYTES // round
+                    + Integer.BYTES // signature array length
+                    + (int) stateSignaturePayload.signature().length()
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/transaction/PayloadWrapper.java:76]: I ask myself if we should do some checks for the lenght if it is > MAX_INT

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/PlatformEvent.java [hunk 2]
```diff
@@ -353,7 +353,7 @@ public void setConsensusTimestampsOnPayloads() {
         if (this.consensusData == NO_CONSENSUS) {
             throw new IllegalStateException("Consensus data must be set");
         }
-        final ConsensusTransactionImpl[] transactions = unsignedEvent.getTransactions();
+        final PayloadWrapper[] transactions = unsignedEvent.getTransactions();
```
Comment (lpetrovic05) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/PlatformEvent.java:356]: if we can get rid of the array and just use a list that would be great

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/internal/EventImpl.java [hunk 3]
```diff
@@ -132,7 +132,7 @@ public Hash getBaseHash() {
     /**
      * @return array of transactions inside this event instance
      */
-    public ConsensusTransactionImpl[] getTransactions() {
```
Comment (lpetrovic05) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/internal/EventImpl.java:135]: we should get rid of the arrays if we can

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/transaction/PayloadWrapper.java [hunk 4]
```diff
@@ -16,21 +16,32 @@
...
      * The consensus timestamp of this transaction, or null if consensus has not yet been reached.
      * NOT serialized and not part of object equality or hash code
      */
     private Instant consensusTimestamp;
     /** An optional metadata object set by the application */
     private Object metadata;
+    /** The protobuf data stored */
+    private final OneOf<PayloadOneOfType> payload;
+
+    public PayloadWrapper(@NonNull final OneOf<PayloadOneOfType> payload) {
```
Comment (lpetrovic05) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/transaction/PayloadWrapper.java:42]: javadoc

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/transaction/PayloadWrapper.java [hunk 5]
```diff
@@ -50,10 +61,25 @@ public void setConsensusTimestamp(@NonNull final Instant consensusTimestamp) {
...
+    public OneOf<PayloadOneOfType> getPayload() {
+        return payload;
+    }
+
+    @Override
+    public int getSize() {
+        if (!isSystem()) {
+            return (int) ((Bytes) payload.as()).length();
+        }
+        return PayloadUtils.getLegacyPayloadSize(payload);
```
[the remark this answers is not in the document]
Comment (timo0) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/transaction/PayloadWrapper.java:106]: I checked the `size()` method of `SwirldsTransaction` and this was the implementation and the implementation in the `PayloadUtils` would add `Integer.BYTES // add the the size of array length field` so the sizes would be 4 grater that way

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/internal/EventImpl.java (hunk 3 above)
Comment (timo0) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/internal/EventImpl.java:135]: done

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/transaction/PayloadWrapper.java (hunk 4 above)
Comment (timo0) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/transaction/PayloadWrapper.java:48]: done

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/DefaultInternalEventValidator.java [hunk 6]
```diff
@@ -177,7 +177,7 @@ private boolean isTransactionByteCountValid(@NonNull final PlatformEvent event)
         int totalTransactionBytes = 0;
         final Iterator<Transaction> iterator = event.transactionIterator();
         while (iterator.hasNext()) {
-            totalTransactionBytes += iterator.next().getSerializedLength();
+            totalTransactionBytes += iterator.next().getSize();
```
[the remark this answers is not in the document]
Comment (timo0) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/DefaultInternalEventValidator.java:180]: done

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/transaction/PayloadWrapper.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (timo0) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/transaction/PayloadWrapper.java]: done

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/PlatformEvent.java (hunk 2 above)
Comment (timo0) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/PlatformEvent.java]: done

**Review by imalygin — APPROVED** [review]

**Review by mxtartaglia-sl — DISMISSED** [review]

**Review by lpetrovic05 — DISMISSED** [review]

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/prehandle/PreHandleWorkflowImplTest.java [hunk 7]
```diff
@@ -191,7 +193,8 @@ void nullConstructorArgsTest() {
     @DisplayName("Null pre-handle args throw NPE")
     @SuppressWarnings("DataFlowIssue") // Suppress the warning about null args
     void nullPreHandleArgsTest() {
-        final List<Transaction> list = List.of(new SwirldTransaction(new byte[10]));
+        final List<Transaction> list = List.of(
+                new PayloadWrapper(new OneOf<>(PayloadOneOfType.APPLICATION_PAYLOAD, Bytes.wrap(new byte[10]))));
```
Comment (netopyr) [inline hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/prehandle/PreHandleWorkflowImplTest.java:197]: Yikes! Is it possible to have a nice factory method?

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/prehandle/PreHandleWorkflowImplTest.java [hunk 8]
```diff
@@ -341,7 +342,8 @@ void preHandlePayerAccountNotFound() throws PreCheckException {
             // (Erin doesn't exist yet)
             final var txInfo = scenario().withPayer(FRANK.accountID()).txInfo();
 
-            final Transaction platformTx = new SwirldTransaction(asByteArray(txInfo.transaction()));
+            final Transaction platformTx = new PayloadWrapper(
+                    new OneOf<>(PayloadOneOfType.APPLICATION_PAYLOAD, Bytes.wrap(asByteArray(txInfo.transaction()))));
```
Comment (netopyr) [inline hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/prehandle/PreHandleWorkflowImplTest.java:346]: I believe the last parameter can be replaced with `Transaction.PROTOBUF.toBytes(txInfo.transaction)`.

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/prehandle/PreHandleWorkflowImplTest.java (hunk 7 above)
Comment (timo0) [reply hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/prehandle/PreHandleWorkflowImplTest.java]: Done

**Review by lpetrovic05 — APPROVED** [review]

**Review by imalygin — APPROVED** [review]

**Review by mhess-swl — APPROVED** [review]
Looks good from services

**Review by tinker-michaelj — APPROVED** [review]
Makes sense to me, ty @timo0 !

**Pull request merged**